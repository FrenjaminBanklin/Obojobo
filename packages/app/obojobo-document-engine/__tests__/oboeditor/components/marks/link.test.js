import { mount } from 'enzyme'
import React from 'react'
import renderer from 'react-test-renderer'
import { Transforms } from 'slate'
jest.mock('slate')
jest.mock('slate-react')

import Link from 'obojobo-document-engine/src/scripts/oboeditor/components/marks/link'

import ModalUtil from 'obojobo-document-engine/src/scripts/common/util/modal-util'
jest.mock('obojobo-document-engine/src/scripts/common/util/modal-util')
jest.mock(
	'obojobo-document-engine/src/scripts/oboeditor/components/node/with-slate-wrapper', 
	() => item => item
)

describe('EditorNav', () => {
	beforeAll(() => {
		document.getSelection = jest.fn()
		document.execCommand = jest.fn()
	})
	beforeEach(() => {
		jest.clearAllMocks()
	})

	test('Link component', () => {
		const component = renderer.create(
			<Link element={{ href: 'mock href' }}/>
		)
		const tree = component.toJSON()
		expect(tree).toMatchSnapshot()
	})

	test('Link component edits', () => {
		const component = mount(
			<Link element={{ href: 'mock href' }}/>
		)

		component.find('button').simulate('click')

		expect(ModalUtil.show).toHaveBeenCalled()
	})

	test('changeLinkValue edits the href with empty href', () => {
		const component = mount(
			<Link element={{ href: 'mock href' }}/>
		)

		component.instance().changeLinkValue(' ')

		expect(Transforms.unwrapNodes).toHaveBeenCalled()
	})

	test('changeLinkValue edits the href', () => {
		const component = mount(
			<Link element={{ href: 'mock href' }}/>
		)

		component.instance().changeLinkValue('mock href')

		expect(Transforms.setNodes).toHaveBeenCalled()
	})
})
