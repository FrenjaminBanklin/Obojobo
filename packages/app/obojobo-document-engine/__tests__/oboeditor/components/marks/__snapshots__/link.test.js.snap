// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditorNav Link component 1`] = `
<span
  className="editor--components--mark--link"
>
  <a
    href="mock href"
    title="mock href"
  />
</span>
`;
